processors/node_visitor.py

```diff
@@ -0,0 +1,146 @@
+import gast as ast
+from typing import List, Dict
+
+
+class ComplexityVisitor(ast.NodeVisitor):
+    def __init__(self):
+        self.operation_complexity = 0
+        self.imports = []
+        self.functions = []
+
+    def visit(self, node):
+        super().visit(node)
+
+        if isinstance(node, (ast.AsyncFor, ast.While, ast.If,
+                             ast.With, ast.AsyncWith)):
+            self.operation_complexity += 0.5
```

Comment:
Here and on on lines 21, 24, 28 (maybe somewhere else) - magic numbers. Maybe better to have object with weights for this numbers